python/ray/tests/test_autoscaler_e2e.py

```diff
@@ -0,0 +1,185 @@
+from functools import partial
+
+import pytest
+import platform
+
+import ray
+from ray.cluster_utils import AutoscalingCluster
+
+
+def test_ray_status_e2e(shutdown_only):
+    cluster = AutoscalingCluster(
+        head_resources={"CPU": 0},
+        worker_node_types={
+            "type-i": {
+                "resources": {"CPU": 1, "fun": 1},
+                "node_config": {},
+                "min_workers": 1,
+                "max_workers": 1,
+            },
+            "type-ii": {
+                "resources": {"CPU": 1, "fun": 100},
+                "node_config": {},
+                "min_workers": 1,
+                "max_workers": 1,
+            },
+        },
+    )
+
+    try:
+        cluster.start()
+        ray.init(address="auto")
+
+        @ray.remote(num_cpus=0, resources={"fun": 2})
+        class Actor:
+            def ping(self):
+                return None
+
+        actor = Actor.remote()
+        ray.get(actor.ping.remote())
+
+        assert "Demands" in subprocess.check_output("ray status", shell=True).decode()
+        assert (
+            "Total Demands"
+            not in subprocess.check_output("ray status", shell=True).decode()
+        )
+        assert (
+            "Total Demands"
+            in subprocess.check_output("ray status -v", shell=True).decode()
+        )
+        assert (
+            "Total Demands"
+            in subprocess.check_output("ray status --verbose", shell=True).decode()
+        )
+    finally:
+        cluster.shutdown()
+
+
+def test_metrics(shutdown_only):
+    cluster = AutoscalingCluster(
+        head_resources={"CPU": 0},
+        worker_node_types={
+            "type-i": {
+                "resources": {"CPU": 1},
+                "node_config": {},
+                "min_workers": 1,
+                "max_workers": 1,
+            },
+            "type-ii": {
+                "resources": {"CPU": 1},
+                "node_config": {},
+                "min_workers": 1,
+                "max_workers": 1,
+            },
+        },
+    )
+
+    try:
+        cluster.start()
+        info = ray.init(address="auto")
+        autoscaler_export_addr = "{}:{}".format(
+            info.address_info["node_ip_address"], AUTOSCALER_METRIC_PORT
+        )
+        print("**************", autoscaler_export_addr, info.address_info)
+
+        @ray.remote(num_cpus=1)
+        class Foo:
+            def ping(self):
+                return True
+
+        wait_for_condition(
+            metric_check_condition(
+                {"autoscaler_cluster_resources": 0, "autoscaler_pending_resources": 0},
+                export_addr=autoscaler_export_addr,
+            )
+        )
+
+        actors = [Foo.remote() for _ in range(2)]
+        ray.get([actor.ping.remote() for actor in actors])
+
+        wait_for_condition(
+            metric_check_condition(
+                {"autoscaler_cluster_resources": 2, "autoscaler_pending_resources": 0},
```

Comment:
What does it mean that we're checking for 2 of `autoscaler_cluster_resources`?
I understand 2 CPU currently present in the cluster, don't understand 2 units of `autoscaler_cluster_resources`.

Reply:
wasn't really sure how to check the metric has a specific tag, i can look into that